build/xtask/src/dist.rs

```diff
@@ -53,45 +55,103 @@ struct PackageConfig {
     remap_paths: BTreeMap<PathBuf, &'static str>,
 }
 
+impl PackageConfig {
+    fn new(app_toml_file: &Path, verbose: bool, edges: bool) -> Result<Self> {
+        let toml = Config::from_file(app_toml_file)?;
+        let dist_dir = Path::new("target").join(&toml.name).join("dist");
+        let app_src_dir = app_toml_file
+            .parent()
+            .ok_or_else(|| anyhow!("Could not get app toml directory"))?;
+
+        Ok(Self {
+            app_toml_file: app_toml_file.to_path_buf(),
+            app_src_dir: app_src_dir.to_path_buf(),
+            toml,
+            verbose,
+            edges,
+            dist_dir,
+            remap_paths: Self::remap_paths()?,
+        })
+    }
+
+    fn dist_file(&self, name: impl AsRef<Path>) -> PathBuf {
+        self.dist_dir.join(name)
+    }
+
+    fn remap_paths() -> Result<BTreeMap<PathBuf, &'static str>> {
+        // Panic messages in crates have a long prefix; we'll shorten it using
+        // the --remap-path-prefix argument to reduce message size.  We'll remap
+        // local (Hubris) crates to /hubris, crates.io to /crates.io, and git
+        // dependencies to /git
+        let mut remap_paths = BTreeMap::new();
+
+        // On Windows, std::fs::canonicalize returns a UNC path, i.e. one
+        // beginning with "\\hostname\".  However, rustc expects a non-UNC
+        // path for its --remap-path-prefix argument, so we use
+        // `dunce::canonicalize` instead
+        let cargo_home = dunce::canonicalize(std::env::var("CARGO_HOME")?)?;
+        let mut cargo_git = cargo_home.clone();
+        cargo_git.push("git");
+        cargo_git.push("checkouts");
+        remap_paths.insert(cargo_git, "/git");
+
+        // This hash is canonical-ish: Cargo tries hard not to change it
+        // https://github.com/rust-lang/cargo/blob/master/src/cargo/core/source/source_id.rs#L607-L630
```

Comment:
how will we know if this changes? purely long error messages, something else? I wonder if there's a good way to be like "this is the place you come fix this"

Reply:
(I know you just moved this code around so I don't think addressing this should block the PR, just wondering now...)

Reply:
Good question – there's no good way to check it automatically, but it should only change when we bump compiler versions, so it's not too hard to check manually.  If we end up writing a playbook for "bumping the Hubris compiler", this should be in there!